src/pages/BillingPage/index.js

```diff
@@ -1,48 +1,93 @@
-import React from 'react';
+import React, { Component } from 'react';
 import get from 'lodash/get';
 import PropTypes from 'prop-types';
 import { Card, Row, Col } from 'antd';
 import { connect } from 'react-redux';
+import Stripe from 'react-stripe-checkout';
+import styled from 'react-emotion';
 import Container from '../../components/Container';
 import BannerHeader from '../../components/Banner/Header';
 import PricingTable from '../../components/PricingTable';
-import { capitalizePlan } from '../../utils/helper';
+import { capitalizePlan, displayErrors } from '../../utils/helper';
 import { getAppPlanByName } from '../../batteries/modules/selectors';
+import { updateAppPaymentMethod } from '../../batteries/modules/actions';
+import Loader from '../../batteries/components/shared/Loader';
 
-const Billing = ({ plan, isOnTrial, planValidity }) => (
-	<React.Fragment>
-		<BannerHeader
-			title="Appbase.io Billing"
-			component={(
-				<Row>
-					<Col>
-						<p>Transparent pricing that scales with you.</p>
-						<p>
-							You are currently on the <strong>{capitalizePlan(plan)}</strong> plan.{' '}
-							{planValidity ? (
-								<span>
-									Valid till{' '}
-									<strong>{new Date(planValidity * 1000).toDateString()}</strong>.
-								</span>
-							) : null}
-						</p>
-						{isOnTrial ? <p>You are currently in trial mode.</p> : null}
-					</Col>
-				</Row>
-			)}
-		/>
-		<Container>
-			<Card bodyStyle={{ padding: 0 }}>
-				<PricingTable />
-			</Card>
-		</Container>
-	</React.Fragment>
-);
+const Link = styled('a')`
+	color: rgb(111, 99, 245);
+	font-weight: 600;
+	text-decoration: underline;
+`;
+
+class Billing extends Component {
+	componentDidUpdate(prevProps) {
+		const { errors } = this.props;
+		displayErrors(errors, prevProps.errors, true);
+	}
+
+	render() {
+		// prettier-ignore
+		const {
+			plan,
+			isOnTrial,
+			planValidity,
+			updatePayment,
+			isLoading,
+		} = this.props;
+		if (isLoading) {
+			return <Loader show message="Updating Payment Method... Please wait!" />;
+		}
+		return (
+			<React.Fragment>
+				<BannerHeader
+					title="Appbase.io Billing"
+					component={(
+						<Row>
+							<Col>
+								<p>Transparent pricing that scales with you.</p>
+								<p>
+									You are currently on the <strong>{capitalizePlan(plan)}</strong>{' '}
+									plan.{' '}
+									{planValidity ? (
+										<span>
+											Valid till{' '}
+											<strong>
+												{new Date(planValidity * 1000).toDateString()}
+											</strong>
+											.
+										</span>
+									) : null}
+								</p>
+								{isOnTrial ? <p>You are currently in trial mode.</p> : null}
+								<Stripe
+									stripeKey="pk_live_ihb1fzO4h1ykymhpZsA3GaQR"
```

Comment:
Create / Use constant defined earlier, so that we can manage stripe testing from single point